### DISCUSSION TITLE
Disable tracing for a specific Java Class or endpoint?

### DISCUSSION THREAD

**Discussion by stevensim226** [open-telemetry/opentelemetry-java-instrumentation#2873] [Q&A]
Hi, 
I'm currently using OpenTelemetry to trace HTTP calls for microservices in my Kubernetes cluster, but unfortunately due to k8s' healthcheck system, every 10 seconds an HTTP call is made into the `/ping` endpoint and it floods my storage for the tracing (I'm using ElasticSearch)

I saw on the manual instrumentation that there is a feature called [suppress class](https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/main/docs/suppressing-instrumentation.md#even-more-fine-grained-control), I tried it w/ auto-instrumentation but it doesn't work (I don't think it's supposed to work either)

Is there a feature similar that exists to what I needed?

**Comment by jkwatson** [comment]
Transferring this discussion to the opentelemetry-java-instrumentation repository, as it applied to instrumentation, not to the API or SDK.

**Comment by ghost** [comment]
@stevensim226 the configuration for suppressing particular classes is meant for automatic instrumentation. That is if the class implementing `/ping` endpoint is added to `otel.javaagent.exclude-classes` it won't be automatically instrumented and won't generate a (server) span.

**Reply by trask** [reply]
In addition to @kubawach's suggestion, it's worth following #1060 which is feature request for ability to exclude health checks
